code/controllers/master.dm

```diff
@@ -161,17 +161,18 @@ GLOBAL_REAL(Master, /datum/controller/master) = new
 
 /datum/controller/master/Recover()
 	var/msg = "## DEBUG: [time2text(world.timeofday)] MC restarted. Reports:\n"
-	for (var/varname in Master.vars)
-		switch (varname)
-			if("name", "tag", "bestF", "type", "parent_type", "vars", "statclick") // Built-in junk.
-				continue
+	var/list/master_attributes = Master.vars
+	var/list/filtered_variables = gather_filtered_variables()
+	for (var/varname in master_attributes)
+		if(filtered_variables.Find(varname)) // Built-in junk.
+			continue
```

Comment:
```suggestion
	var/list/master_attributes = Master.vars
	var/list/filtered_variables = list(NAMEOF(src, name), NAMEOF(src, tag), NAMEOF(src, type), NAMEOF(src, parent_type) NAMEOF(src, vars), NAMEOF(src, statclick))
	for (var/varname in master_attributes - filtered_variables)
```

NAMEOF returns the string version of the varname.

```dm
world.log << json_encode(NAMEOF(src, statclick)))
```

returns `"statclick"` ie, NAMEOF even adds the "" for you.